src/legacy/core_plugins/timelion/public/plugin.ts

```diff
@@ -76,7 +68,7 @@ export class TimelionPlugin implements Plugin<Promise<void>, void> {
     this.registerPanels(dependencies);
 
     expressions.registerFunction(() => getTimelionVisualizationConfig(dependencies));
-    visualizations.types.createBaseVisualization(getTimelionVisualization(dependencies));
+    visualizations.types.createBaseVisualization(await getTimelionVisualization(dependencies));
```

Comment:
I'm not 100% sure whether that's a good idea or not. It basically means the visualization will be registered in the start phase, because the promise chain is connected to `core.getStartServices`.  @lizozom is that a problem? If it is, then the promise should be awaited in the `TimelionOptions` component and delay rendering there.

Reply:
The pattern described by @lizozom here https://github.com/elastic/kibana/pull/52990#discussion_r358930306 was used instead

Reply:
That looks good to me - we use this pattern in several apps (e.g. home and discover)